### PR TITLE
Debug read batch unbounded

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

**Comment by je-ik** [comment]
Closes O2-Czech-Republic#277

**Review by mareksabo — COMMENTED** [review]
Looks good, I just have a few nits/questions.

File: beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BatchLogRead.java [hunk 1]
```diff
@@ -147,37 +156,49 @@ public ProcessContinuation process(
 
       watermarkEstimator.setWatermark(tracker.currentRestriction().getMinTimestamp());
 
-      while (!tracker.currentRestriction().isFinished()) {
```
[the remark this answers is not in the document]
Comment (je-ik) [reply beam/core/src/main/java/cz/o2/proxima/beam/direct/io/BatchLogRead.java:150]: It does not cycle anymore, but return ProcessContinuation.resume(). That is to enable checkpointing.

**Comment by mareksabo** [comment]
LGTM 👍

**Pull request merged**